### PR TITLE
test: move mistral / mixtral test cases in QA test list into the new accuracy test suite

### PR DESCRIPTION
**Pull request by crazydemo**
This PR moves alll the summarization / mmlu task in examples/test_mistral.py and  examples/test_mixtral.py into the new accuracy test suite.

- [x] wait for local check
- [x] adjust accuracy threshold

### CODE REVIEW

File: tensorrt_llm/llmapi/llm_utils.py [hunk 1]
```diff
@@ -418,6 +418,24 @@ def _update_from_hf_quant_config(self) -> bool:
...
+                # AWQ. Mistral-8x7b-v0.1-AWQ
+                elif hf_quant_config.get("quant_method") == "awq":
+                    logger.info(
+                        "AWQ quantization detected in config.json, setting up appropriate quantization parameters."
+                    )
+                    bits = hf_quant_config.get("bits", 4)
+                    if bits == 4:
+                        quant_config.quant_algo = QuantAlgo.W4A16_AWQ
+                    elif bits == 8:
+                        quant_config.quant_algo = QuantAlgo.W4A8_AWQ
```
Comment (syuoni) [inline tensorrt_llm/llmapi/llm_utils.py]: What does the `bits` mean? Is it for weight or activation? I'm confused how the `bits` is mapped to `QuantAlgo`.

File: tests/integration/defs/accuracy/references/cnn_dailymail.yaml [hunk 2]
```diff
@@ -148,15 +148,65 @@ meta-llama/Llama-3.3-70B-Instruct:
...
+    accuracy: 24.0
+NeMo/Mistral_NeMo_Minitron_8B_Instruct:
+  - quant_algo: FP8
+    accuracy: 32.553
 mistralai/Mixtral-8x7B-v0.1:
   - accuracy: 28.810
+  - extra_acc_spec: plugin;beam_width=4
+    accuracy: 29.812
+  - extra_acc_spec: ootb_except_mha;beam_width=4
+    accuracy: 29.981
```
Comment (syuoni) [inline tests/integration/defs/accuracy/references/cnn_dailymail.yaml]: It's fine to use `extra_acc_spec` to specify `beam_width`, since it can be well justified why beam search affects generation results.

However, we normally expect the plugin- and OOTB-implemented models should have same accuracy, so they should share a reference accuracy. Similar rules apply to `pp_reduce_scatter` below.

BTW, for a given model, we expect justified accuracy difference for
* different dtypes
* different quantization methods
* different speculative decoding methods
So they are listed in [acc_specs](https://github.com/NVIDIA/TensorRT-LLM/blob/main/tests/integration/defs/accuracy/accuracy_core.py#L91-L96)

The last one, `extra_acc_spec` is used for anything that falls out of the above categories. For example, `beam_width`. Please do not over-use `extra_acc_spec`.

File: tests/integration/defs/accuracy/references/cnn_dailymail.yaml [hunk 3]
```diff
@@ -148,15 +148,65 @@ meta-llama/Llama-3.3-70B-Instruct:
...
+    accuracy: 28.368
+  - extra_acc_spec: beam_width=2
+    accuracy: 27.663
+  - extra_acc_spec: beam_width=1
+    accuracy: 25.604
+  - quant_algo: W8A8_SQ_PER_CHANNEL_PER_TOKEN_PLUGIN
+    extra_acc_spec: beam_width=4
+    accuracy: 27.745
+  - quant_algo: FP8
+    extra_acc_spec: beam_width=4
```
Comment (syuoni) [inline tests/integration/defs/accuracy/references/cnn_dailymail.yaml]: Is it necessary to test so many beam search cases?

When validating a model's accuracy, we normally expect the accuracy should be OK without beam search; unless the test aims to validate the accuracy of the beam search feature.

File: tests/integration/defs/accuracy/test_cli_flow.py [hunk 4]
```diff
@@ -114,8 +114,7 @@ def test_fp8(self):
 
     @skip_pre_ada
     def test_fp8_lm_head(self):
-        self.run(quant_algo=QuantAlgo.FP8,
-                 extra_convert_args=["--quantize_lm_head"])
+        self.run(quant_algo=QuantAlgo.FP8, t=["--quantize_lm_head"])
```
Comment (syuoni) [inline tests/integration/defs/accuracy/test_cli_flow.py]: Is this modified by mistake?

File: tests/integration/defs/accuracy/test_cli_flow.py [hunk 5]
```diff
@@ -766,6 +765,45 @@ def test_cyclic_kv_cache_beam_search(self):
...
+                 extra_build_args=["--gemm_plugin=auto", "--max_beam_width=4"],
+                 extra_summarize_args=["--num_beams=4"])
+        import gc
+
+        import torch
+        for num_beams in [1, 2]:
+            gc.collect()
+            torch.cuda.empty_cache()
+            self.extra_summarize_args = [f"--num_beams={num_beams}"]
+            self.evaluate()
```
Comment (syuoni) [inline tests/integration/defs/accuracy/test_cli_flow.py:815]: This test seems not finished?

File: tests/integration/defs/accuracy/test_cli_flow.py [hunk 6]
```diff
@@ -816,6 +923,73 @@ def test_nvfp4_prequantized(self, mocker):
...
+            build_args.extend([
+                "--gemm_plugin=disable"
+                if fp4_type == "ootb" else "--gemm_plugin=nvfp4"
+            ])
+        if fp4_type == "plugin":
+            build_args.extend([
+                "--use_paged_context_fmha=enable",
+                "--use_fp8_context_fmha=enable"
+            ])
+        if fp4_type != "disable":
```
Comment (syuoni) [inline tests/integration/defs/accuracy/test_cli_flow.py]: Please create a new test case if it's not FP4.

Please do not over-use `pytest.mark.parametrize`, especially parametrizing a test to run something not reflecting the test name anymore.

File: tests/integration/defs/accuracy/test_llm_api.py [hunk 7]
```diff
@@ -37,6 +38,74 @@ def test_fp8_rowwise(self):
             task.evaluate(llm)
 
 
+class TestMistral7B_0_3(LlmapiAccuracyTestHarness):
+    MODEL_NAME = "mistralai/Mistral-7B-v0.3"
+    MODEL_PATH = f"{llm_models_root()}/Mistral-7B-Instruct-v0.3"
```
Comment (syuoni) [inline tests/integration/defs/accuracy/test_llm_api.py:43]: If the model path is correct, the model name should be `mistralai/Mistral-7B-Instruct-v0.3` (https://huggingface.co/mistralai/Mistral-7B-Instruct-v0.3). 

Note that models with `Instruct`/`Chat` in the name are different from the base model, and the accuracies are different.

File: tests/integration/defs/accuracy/test_llm_api.py [hunk 8]
```diff
@@ -37,6 +38,74 @@ def test_fp8_rowwise(self):
...
+            quant_config = QuantConfig(quant_algo=QuantAlgo.W4A16)
+        elif quant == 'int4_awq':
+            quant_config = QuantConfig(quant_algo=QuantAlgo.W4A16_AWQ)
+        elif quant == 'int8_awq':
+            quant_config = QuantConfig(quant_algo=QuantAlgo.W4A8_AWQ)
+
+        build_config = BuildConfig()
+        build_config.max_batch_size = 1
+        build_config.max_input_len = 1900
+        build_config.plugin_config.paged_kv_cache = True
```
Comment (syuoni) [inline tests/integration/defs/accuracy/test_llm_api.py]: I understand you are moving old tests to here. There are some legacy reasons why we didn't use default values for BuildConfig.

Currently, I believe we normally don't need to change BuildConfig when using LLM API.

File: tests/integration/defs/accuracy/test_llm_api.py [hunk 9]
```diff
@@ -37,6 +38,74 @@ def test_fp8_rowwise(self):
...
+                 quant_config=quant_config) as llm:
+            task = CnnDailymail(self.MODEL_NAME)
+            task.evaluate(llm)
+            task = MMLU(self.MODEL_NAME)
+            task.evaluate(llm)
+
+
+class TestMistral_Nemo_12B_Base(LlmapiAccuracyTestHarness):
+    MODEL_NAME = "NeMo/Mistral_Nemo_12B_Base"
+    MODEL_PATH = f"{llm_models_root()}/Mistral-Nemo-Base-2407"
```
Comment (syuoni) [inline tests/integration/defs/accuracy/test_llm_api.py:69]: I suppose the model name should be `mistralai/Mistral-Nemo-Base-2407`. Please verify the model name on HF hub.

File: tests/integration/defs/accuracy/test_llm_api.py [hunk 10]
```diff
@@ -37,6 +38,74 @@ def test_fp8_rowwise(self):
...
+
+    def test_fp8(self):
+        quant_config = QuantConfig(quant_algo=QuantAlgo.FP8,
+                                   kv_cache_quant_algo=QuantAlgo.FP8)
+
+        build_config = BuildConfig()
+        build_config.plugin_config._gemm_plugin = "auto"
```
Comment (syuoni) [inline tests/integration/defs/accuracy/test_llm_api.py]: Unless this test aims to test the plugin implementation, we should just use the default settings.

**Review by syuoni — COMMENTED** [review]
Thanks for the efforts to migrate these tests!

File: tests/integration/defs/accuracy/test_cli_flow.py (hunk 5 above)
Comment (crazydemo) [reply tests/integration/defs/accuracy/test_cli_flow.py:787]: this test have already been finished, it aims to use the same engine to run with different `num_beams`

File: tests/integration/defs/accuracy/test_llm_api.py (hunk 7 above)
Comment (crazydemo) [reply tests/integration/defs/accuracy/test_llm_api.py:43]: fixed.

File: tests/integration/defs/accuracy/test_llm_api.py (hunk 9 above)
Comment (crazydemo) [reply tests/integration/defs/accuracy/test_llm_api.py:69]: fixed.

File: tensorrt_llm/llmapi/llm_utils.py (hunk 1 above)
Comment (crazydemo) [reply tensorrt_llm/llmapi/llm_utils.py]: removed these code. As prequantized models are tested with `test_cli_flow`.

File: tests/integration/defs/accuracy/references/cnn_dailymail.yaml (hunk 2 above)
Comment (crazydemo) [reply tests/integration/defs/accuracy/references/cnn_dailymail.yaml]: Thanks for your clarification. fixed.

File: tests/integration/defs/accuracy/references/cnn_dailymail.yaml (hunk 3 above)
Comment (crazydemo) [reply tests/integration/defs/accuracy/references/cnn_dailymail.yaml]: reduce the beam search test.

File: tests/integration/defs/accuracy/test_cli_flow.py (hunk 4 above)
Comment (crazydemo) [reply tests/integration/defs/accuracy/test_cli_flow.py]: fixed.

File: tests/integration/defs/accuracy/test_cli_flow.py (hunk 6 above)
Comment (crazydemo) [reply tests/integration/defs/accuracy/test_cli_flow.py]: fixed.

File: tests/integration/defs/accuracy/test_llm_api.py (hunk 8 above)
Comment (crazydemo) [reply tests/integration/defs/accuracy/test_llm_api.py]: use default config instead.

File: tests/integration/defs/accuracy/test_llm_api.py (hunk 10 above)
Comment (crazydemo) [reply tests/integration/defs/accuracy/test_llm_api.py]: fixed.

File: tests/integration/defs/accuracy/references/cnn_dailymail.yaml [hunk 11]
```diff
@@ -148,15 +148,57 @@ meta-llama/Llama-3.3-70B-Instruct:
...
+    accuracy: 32.553
 mistralai/Mixtral-8x7B-v0.1:
   - accuracy: 28.810
   - quant_algo: NVFP4
     kv_cache_quant_algo: FP8
     accuracy: 29.733
   - quant_algo: FP8
     kv_cache_quant_algo: FP8
-    accuracy: 27.109
+    accuracy: 24.648
```
Comment (syuoni) [inline tests/integration/defs/accuracy/references/cnn_dailymail.yaml:185]: Why should we lower this reference accuracy?

File: tests/integration/defs/accuracy/references/mmlu.yaml [hunk 12]
```diff
@@ -31,14 +31,26 @@ meta-llama/Llama-3.3-70B-Instruct:
...
+    accuracy: 60.04
 mistralai/Mixtral-8x7B-v0.1:
   - accuracy: 71.35
   - quant_algo: FP8
     kv_cache_quant_algo: FP8
     accuracy: 71.27
   - quant_algo: NVFP4
     kv_cache_quant_algo: FP8
-    accuracy: 69.64
+    accuracy: 68.0
```
Comment (syuoni) [inline tests/integration/defs/accuracy/references/mmlu.yaml:48]: Same here, why should we lower this reference accuracy?

File: tests/integration/defs/accuracy/test_cli_flow.py (hunk 5 above)
Comment (syuoni) [reply tests/integration/defs/accuracy/test_cli_flow.py:788]: OK. Thanks for the explanation.

File: tests/integration/defs/accuracy/test_llm_api.py [hunk 13]
```diff
@@ -49,6 +102,49 @@ def test_tp2(self):
...
+        with LLM(self.MODEL_PATH) as llm:
+            task = MMLU(self.MODEL_NAME)
+            task.evaluate(llm)
+
+    @skip_pre_ada
+    def test_fp8_prequantized_tp4(self, mocker):
+        mocker.patch.object(
+            self.__class__, "MODEL_PATH",
+            f"{llm_models_root()}/Mixtral-8x7B-Instruct-v0.1-fp8")
+        with LLM(self.MODEL_PATH, tensor_parallel_size=4) as llm:
```
Comment (syuoni) [inline tests/integration/defs/accuracy/test_llm_api.py:144]: We don't need to use mock in LLM API style tests. Just like

```python
with LLM(f"{llm_models_root()}/Mixtral-8x7B-Instruct-v0.1-fp8", tensor_parallel_size=4) as llm:
    task = MMLU(self.MODEL_NAME)
    task.evaluate(llm)
```

File: tests/integration/defs/accuracy/test_llm_api.py [hunk 14]
```diff
@@ -49,6 +102,49 @@ def test_tp2(self):
...
+                 tensor_parallel_size=2) as llm:
+            task = CnnDailymail(self.MODEL_NAME)
+            task.evaluate(llm)
+
+    @skip_pre_blackwell
+    def test_fp4_prequantized(self, mocker):
+        mocker.patch.object(
+            self.__class__, "MODEL_PATH",
+            f"{llm_models_root()}/nvfp4-quantized/Mixtral-8x7B-Instruct-v0.1")
+        with LLM(self.MODEL_PATH) as llm:
```
Comment (syuoni) [inline tests/integration/defs/accuracy/test_llm_api.py:135]: We don't need to use mock in LLM API style tests. See another comment.

**Comment by syuoni** [comment]
Hi @crazydemo ,

Thanks again for the efforts to migrate these tests. I'm not asking for changing this PR, but in the future, if possible, we should directly migrate the tests to be LLM API style, i.e., those in `test_llm_api.py` and `test_llm_api_pytorch.py`.

File: tests/integration/defs/accuracy/references/cnn_dailymail.yaml (hunk 11 above)
Comment (crazydemo) [reply tests/integration/defs/accuracy/references/cnn_dailymail.yaml]: revert to the original data.

File: tests/integration/defs/accuracy/references/mmlu.yaml (hunk 12 above)
Comment (crazydemo) [reply tests/integration/defs/accuracy/references/mmlu.yaml]: revert to the original data.

File: tests/integration/defs/accuracy/test_llm_api.py (hunk 13 above)
Comment (crazydemo) [reply tests/integration/defs/accuracy/test_llm_api.py]: fixed.

File: tests/integration/defs/accuracy/test_llm_api.py (hunk 14 above)
Comment (crazydemo) [reply tests/integration/defs/accuracy/test_llm_api.py]: fixed.

**Comment by crazydemo** [comment]
> Hi @crazydemo ,
> 
> Thanks again for the efforts to migrate these tests. I'm not asking for changing this PR, but in the future, if possible, we should directly migrate the tests to be LLM API style, i.e., those in `test_llm_api.py` and `test_llm_api_pytorch.py`.

Thanks for your review and suggestion. I will add new test cases to `test_llm_api.py` and `test_llm_api_pytorch.py`. Since we have examples cases in `examples/models/core`, we should cover these cli cases to ensure them work well.

**Review by LarryXFly — APPROVED** [review]

**Comment by crazydemo** [comment]
/bot run --stage-list "A10-PackageSanityCheck-manylinux-py312"

**Comment by crazydemo** [comment]
/bot run --stage-list "A10-PackageSanityCheck-manylinux-py312"

**Comment by crazydemo** [comment]
/bot run --stage-list "A10-PackageSanityCheck-manylinux-py312"

File: tests/integration/defs/accuracy/test_cli_flow.py [hunk 15]
```diff
@@ -835,6 +910,47 @@ def test_fp8_tp2pp2_manage_weights(self):
...
+    @pytest.mark.skip_less_device(2)
+    @pytest.mark.skip_less_device_memory(80000)
+    def test_weight_only_int4_tp2(self):
+        self.run(quant_algo=QuantAlgo.W4A16,
+                 tp_size=2,
+                 extra_build_args=["--gemm_plugin=auto"])
+
+    @skip_post_blackwell
+    def test_int4_awq_prequantized(self):
+        self.run(quant_algo=QuantAlgo.W4A16_AWQ)
```
Comment (syuoni) [inline tests/integration/defs/accuracy/test_cli_flow.py:922]: If I'm not wrong, this test will call `modelopt` to do quantization instead of using pre-quantized ckpt. Please double check it.

**Review by syuoni — APPROVED** [review]
LGTM, thanks!

File: tests/integration/defs/accuracy/test_cli_flow.py (hunk 15 above)
Comment (crazydemo) [reply tests/integration/defs/accuracy/test_cli_flow.py]: yes, you are right. This is duplicated, removed.

**Review by LarryXFly — APPROVED** [review]
